### PR TITLE
Add a tool to suggest increasing the severity of performance-impacting bugs

### LINKED ISSUE #1709: Suggest increasing the severity for bugs that their `Performance Impact` flag is set to `high`
> Suggested by @marco-c in Matrix 

Example: https://bugzilla.mozilla.org/show_bug.cgi?id=1741865

Bugzilla query: https://bugzilla.mozilla.org/buglist.cgi?f1=cf_performance_impact&o1=equals&v1=high

### PR DESCRIPTION
**Pull request by suhaibmujahid**
Resolves #1709

# Autonag wiki page
```wiki
{{AutonagRule
  | Rule name = Severity of performance impacting bugs
  | Purpose   = Consistency and help getting traction on bugs with a high impact on performance
  | Action    = Needinfo the assignee or the triage owner (if not assigned) to increase the severity
  | Source    = severity_high_performance_impact.py 
}}
```

## Checklist

- [ ] Type annotations added to new functions
- [ ] Docs added to functions touched in main classes
- [x] Dry-run produced the expected results
- [ ] The [`to-be-announced`](https://github.com/mozilla/relman-auto-nag/labels/to-be-announced) tag added if this is worth announcing

### CODE REVIEW

**Comment by suhaibmujahid** [comment]
# Dry-run

```
2022-11-24 20:56:10,145 - INFO - Run tool severity_high_performance_impact.py
2022-11-24 20:56:13,959 - INFO - Tool severity_high_performance_impact.py has finished.
```

<p>The following bugs have discrepancy between the bug severity and the performance impact evaluation. Owners got needinfo requests to bump the severity:
<table style="border:1px solid black;border-collapse:collapse;" border="1">
<thead>
<tr>
<th>Bug</th><th>Summary</th><th>Severity</th>
</tr>
</thead>
<tbody>
<tr bgcolor="#E0E0E0">
<td>
<a href="https://bugzilla.mozilla.org/show_bug.cgi?id=1789268">1789268</a>
</td>
<td>
DrawSurfaceWithShadow in canvas is very slow
</td>
<td>
S3
</td>
</tr>
<tr >
<td>
<a href="https://bugzilla.mozilla.org/show_bug.cgi?id=1771902">1771902</a>
</td>
<td>
High CPU use to make cursors blink in google docs in fully occluded windows
</td>
<td>
S3
</td>
</tr>
<tr bgcolor="#E0E0E0">
<td>
<a href="https://bugzilla.mozilla.org/show_bug.cgi?id=1761734">1761734</a>
</td>
<td>
Slow drawing of this PDF file
</td>
<td>
S3
</td>
</tr>
<tr >
<td>
<a href="https://bugzilla.mozilla.org/show_bug.cgi?id=1760045">1760045</a>
</td>
<td>
Slow HTMLElement.focus calls on matrix-react benchmark
</td>
<td>
S3
</td>
</tr>
<tr bgcolor="#E0E0E0">
<td>
<a href="https://bugzilla.mozilla.org/show_bug.cgi?id=1755864">1755864</a>
</td>
<td>
Process using 100% of the CPU on the main thread recording infinite recursion error stacks
</td>
<td>
S4
</td>
</tr>
<tr >
<td>
<a href="https://bugzilla.mozilla.org/show_bug.cgi?id=1752841">1752841</a>
</td>
<td>
Pretty bad Google Docs perf with GPU canvas.
</td>
<td>
S3
</td>
</tr>
<tr bgcolor="#E0E0E0">
<td>
<a href="https://bugzilla.mozilla.org/show_bug.cgi?id=1745512">1745512</a>
</td>
<td>
Significant performance impact when using Webpagetest and ES Modules
</td>
<td>
S3
</td>
</tr>
<tr >
<td>
<a href="https://bugzilla.mozilla.org/show_bug.cgi?id=1741865">1741865</a>
</td>
<td>
Firefox loads tabs slowly after starting and update to new version
</td>
<td>
S3
</td>
</tr>
<tr bgcolor="#E0E0E0">
<td>
<a href="https://bugzilla.mozilla.org/show_bug.cgi?id=1725781">1725781</a>
</td>
<td>
SAP application very slow since 72.0b1, with deeply nested percent-height tables
</td>
<td>
S3
</td>
</tr>
<tr >
<td>
<a href="https://bugzilla.mozilla.org/show_bug.cgi?id=1694048">1694048</a>
</td>
<td>
Reddit Become Slower
</td>
<td>
S3
</td>
</tr>
<tr bgcolor="#E0E0E0">
<td>
<a href="https://bugzilla.mozilla.org/show_bug.cgi?id=1686592">1686592</a>
</td>
<td>
Responsiveness: On Twitter, switching from Home to Explore takes 2.5x as long as in Chrome
</td>
<td>
S3
</td>
</tr>
<tr >
<td>
<a href="https://bugzilla.mozilla.org/show_bug.cgi?id=1677509">1677509</a>
</td>
<td>
SendReceiveMouseInputEvent often blocks the main thread of the parent process
</td>
<td>
S3
</td>
</tr>
<tr bgcolor="#E0E0E0">
<td>
<a href="https://bugzilla.mozilla.org/show_bug.cgi?id=1664492">1664492</a>
</td>
<td>
Extremely long DNS resolution times with Fenix
</td>
<td>
S3
</td>
</tr>
<tr >
<td>
<a href="https://bugzilla.mozilla.org/show_bug.cgi?id=1647153">1647153</a>
</td>
<td>
Slow reloading of Pocket Explore on Moto G (XT1031)
</td>
<td>
S3
</td>
</tr>
<tr bgcolor="#E0E0E0">
<td>
<a href="https://bugzilla.mozilla.org/show_bug.cgi?id=1602842">1602842</a>
</td>
<td>
OSXDisplay::EnableVsync hangs in CVDisplayLinkCreateWithActiveCGDisplays
</td>
<td>
S3
</td>
</tr>
<tr >
<td>
<a href="https://bugzilla.mozilla.org/show_bug.cgi?id=1595680">1595680</a>
</td>
<td>
Slightly different values in display lists causing extra picture cache invalidations.
</td>
<td>
S3
</td>
</tr>
<tr bgcolor="#E0E0E0">
<td>
<a href="https://bugzilla.mozilla.org/show_bug.cgi?id=1550754">1550754</a>
</td>
<td>
ContentParent::KillHard janks the main thread
</td>
<td>
S3
</td>
</tr>
<tr >
<td>
<a href="https://bugzilla.mozilla.org/show_bug.cgi?id=1389712">1389712</a>
</td>
<td>
switching to a Google Slides tab is slow (long repaint)
</td>
<td>
S3
</td>
</tr>
<tr bgcolor="#E0E0E0">
<td>
<a href="https://bugzilla.mozilla.org/show_bug.cgi?id=1363383">1363383</a>
</td>
<td>
Channel switching on Slack is slow
</td>
<td>
S3
</td>
</tr>
<tr >
<td>
<a href="https://bugzilla.mozilla.org/show_bug.cgi?id=1356686">1356686</a>
</td>
<td>
OMT brotli decompression
</td>
<td>
S3
</td>
</tr>
</tbody>
</table>
</p>

## Comment example - [bug 1752841](https://bugzilla.mozilla.org/show_bug.cgi?id=1752841)

The severity field for this bug is set to S3. However, the `Performance Impact` field flags this bug as having a high impact on the performance.
:lsalzman, could you consider increasing the severity of this performance-impacting bug?

For more information, please visit [auto_nag documentation](https://wiki.mozilla.org/Release_Management/autonag#severity_high_performance_impact.py).

File: templates/severity_high_performance_impact_needinfo.txt [hunk 1]
```diff
@@ -0,0 +1,4 @@
+The severity field for this bug is set to {{ extra[bugid]["severity"] }}. However, the `Performance Impact` field flags this bug as having a high impact on the performance.
+:{{ nickname }}, could you consider increasing the severity of this performance-impacting bug?
```
Comment (marco-c) [inline templates/severity_high_performance_impact_needinfo.txt:2]:
```suggestion
:{{ nickname }}, could you consider increasing the severity of this performance-impacting bug?
```
Maybe we could say something like "Or, if you think the impact is lower than previously thought, could you request retriage from the performance team?"

Comment (davehunt) [reply templates/severity_high_performance_impact_needinfo.txt:2]: I like @marco-c's suggestion about requesting retriage.

Comment (suhaibmujahid) [reply templates/severity_high_performance_impact_needinfo.txt]: Done in 588a279b0b993d40548e76ad937324ad2ffc5f89

**Review by marco-c — APPROVED** [review]

**Pull request merged**